memory/persona.py

```
@@ -0,0 +1,556 @@
+            return persona.setdefault('user', {}).setdefault('facts', [])
+        elif entity == 'ai':
+            return persona.setdefault('ai', {}).setdefault('facts', [])
+        else:
+            return persona.setdefault('relationship', {}).setdefault('dynamics', [])
+
+    @staticmethod
+    def _texts_may_contradict(old_text: str, new_text: str) -> bool:
+        """Lightweight keyword-overlap heuristic for contradiction detection."""
+        if not old_text or not new_text:
+            return False
+        old_words = set(old_text.replace('，', ' ').replace('。', ' ').split())
```

Ruff (0.15.7)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)